xmtp_mls/src/client.rs

```diff
@@ -593,32 +600,41 @@ where
         Ok(groups)
     }
 
-    pub async fn sync_all_groups(&self, groups: Vec<MlsGroup>) -> Result<(), GroupError> {
+    /// Sync all groups for the current user and return the number of groups that were synced.
+    /// Only active groups will be synced.
+    pub async fn sync_all_groups(&self, groups: Vec<MlsGroup>) -> Result<usize, GroupError> {
         use scoped_futures::ScopedFutureExt;
 
         // Acquire a single connection to be reused
         let provider: XmtpOpenMlsProvider = self.mls_provider()?;
 
+        let active_group_count = Arc::new(AtomicUsize::new(0));
+
         let sync_futures: Vec<_> = groups
             .into_iter()
             .map(|group| {
                 async {
                     // create new provider ref that gets moved, leaving original
                     // provider alone.
                     let provider_ref = &provider;
+                    let active_group_count = Arc::clone(&active_group_count);
```

Comment:
Ah thanks for this fix. I had a thought this might be an issue and asked in slack. But should have validated. 🙏